lib/parallel-fields-extractor.js

```diff
@@ -0,0 +1,14 @@
+module.exports = (elasticSearchResponse) => {
+  elasticSearchResponse.hits.hits.forEach((bib) => {
+    if (bib._source.parallelDisplayField) {
+      bib._source.parallelDisplayField.forEach((field) => {
+        const capitalizedFieldName = field.fieldName.charAt(0).toUpperCase() + field.fieldName.slice(1)
+        bib._source[`parallel${capitalizedFieldName}`] = []
+        bib._source[`parallel${capitalizedFieldName}`][field.index] = field.value
+      })
+      delete bib._source.parallelDisplayField
```

Comment:
Seems potentially error prone. Would prefer to clone the bib and delete from the clone rather than mutate the original bib.

Reply:
In general I agree we should not mutate like this, but I think the memory cost savings of _not_ cloning potentially large bib documents and the fact that `bib._source.parallelDisplayField` _only_ serves to populate these dynamically generated parallel fields justifies it, personally.